### ISSUE TITLE
languages.yaml should accept compilers in path and not hardwire absolute paths

### ISSUE THREAD

**Issue by godmar** [Kattis/problemtools#103]
Is there a reason for why languages.yaml hardwires the absolute path of all compilers?

It causes issues when I need to use a particular version of gcc/g++ on systems where /usr/bin/g++ is not the version problemtools wants, such as CentOS.  On those systems, you start a shell such that `/opt/rh/devtoolset-7/root/usr/bin/g++` is in the path before `/usr/bin/g++` using [RedHat's scl-utils](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Developer_Guide/scl-utils.html).

problemtools completely breaks that with its assumption that the [absolute path to an executable be provided in languages.yaml](https://github.com/Kattis/problemtools/blob/develop/problemtools/run/source.py#L100-L104)

Do you want it this way, or would you entertain a PR to accept commands from the path in addition?

**Comment by austrin** [comment]
> Is there a reason for why languages.yaml hardwires the absolute path of all compilers?

Yes, but it's not very good (or it used to be good but has played out, or I forgot what it was, doesn't really matter).

A PR fixing the behaviour you mention that verifyproblem breaks if absolute paths are not provided would definitely be welcome.

Whether to also change the defaults in problem.yaml or not, I'd like to mull over this a little while first, but given the recently added https://github.com/Kattis/problemtools#configuration this should also be relatively easy for you to work around once the first issue has been fixed.
